frontend/src/RecordingPageComp/InstrumentSelectionComp.jsx

```diff
@@ -14,8 +18,15 @@ function InstrumentSelection({ midiSounds, setSelectedInstrument }) {
   }
 
   const handleSelectInstrument = (index) => {
-    setSelectedIdx(index); // Set the selected index
+    if (index === selectedIdx) setSelectedIdx(null); // Set the selected index
+    else setSelectedIdx(index);
+
     setSelectedInstrument(index); // Perform any additional actions
+    setIsLoadingInstrument(true); // Start loading
+
+    setTimeout(() => {
+      setIsLoadingInstrument(false);
+    }, 1000);
```

Comment:
Do you still want to load if the selected index is the same? Also I think you want to use setTimeout within an useEffect hook as using it within the component itself may degrade performance? See this article for more details: https://codedamn.com/news/reactjs/how-to-use-settimeout-in-react-complete-guide-with-examples